src/AppInstallerCLICore/Command.cpp

```diff
@@ -740,6 +740,26 @@ namespace AppInstaller::CLI
             }
         }
 
+        if (execArgs.Contains(Execution::Args::Type::SourceTrustLevel))

```

Comment:
This check is specific to "source add", move to ValidateArguments of SourceAdd command